File: Java/DesignPatterns/src/test/java/designPatterns/behavioral/observer/MainObserverTest.java
package designPatterns.behavioral.observer;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainObserverTest {
  private Topic topic;
  private Observer obj1;
  private Observer obj2;
  private Observer obj3;

  @BeforeEach
  void setUp() {
    topic = new Topic();
    obj1 = new TopicSubscriber("Subscriber1");
    obj2 = new TopicSubscriber("Subscriber2");
    obj3 = new TopicSubscriber("Subscriber3");
    topic.register(obj1);
    topic.register(obj2);
    topic.register(obj3);
  }

  @Test
  void equalsTest() {
    assertThat(topic.postMessage("Hello Subscribers!")).isEqualTo("Message Posted to Topic: Hello Subscribers!");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/behavioral/observer/MainObserverTest.kt
package behavioral.observer

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainObserverTest {
  private lateinit var topic: Topic
  private lateinit var obj1: Observer
  private lateinit var obj2: Observer
  private lateinit var obj3: Observer

  @BeforeEach
  internal fun setUp() {
    topic = Topic()
    obj1 = TopicSubscriber("Subscriber1")
    obj2 = TopicSubscriber("Subscriber2")
    obj3 = TopicSubscriber("Subscriber3")
    topic.register(obj1)
    topic.register(obj2)
    topic.register(obj3)
  }

  @Test
  internal fun equalsTest() {
    assertThat(topic.postMessage("Hello Subscriber!")).isEqualTo("Message Posted to Topic: Hello Subscriber!")
  }
}